src/lib.rs

```diff
@@ -1240,21 +1248,146 @@ fn to_lowercase(name: &Ident) -> Ident {
     Ident::new(&s, name.span())
 }
 
-/// Create a tokenstream for a lifetime bound/application if it's needed
-fn lifetime(needs_lifetime: bool) -> proc_macro2::TokenStream {
-    if needs_lifetime {
-        quote! { <'a> }
-    } else {
-        quote! {}
+/// Validate generic parameters on a relation
+/// Checks for unsupported features and provides helpful error messages
+fn validate_generic_params(relation: &Relation) {
+    // Type parameters are now supported!
```

Comment:
too much information haha
```suggestion
    // Type parameters are supported
```